test/index.js

```diff
@@ -2989,6 +3003,18 @@ describe('def', function() {
 
     eq(bimap.toString(), 'bimap :: Bifunctor f => (a -> b) -> (c -> d) -> f a c -> f b d');
     eq(bimap(length, Math.sqrt, Pair('Sanctuary', 25)), Pair(9, 5));
+
+    throws(function() { bimap(length, Math.sqrt, Pair(['foo', true, 42], null)); },
+           TypeError,
+           'Type-variable constraint violation\n' +
+           '\n' +
+           'bimap :: Bifunctor f => (a -> b) -> (c -> d) -> f a c -> f b d\n' +
+           '                                                  ^\n' +
+           '                                                  1\n' +
+           '\n' +
+           '1)  ["foo", true, 42] :: Array ???\n' +
+           '\n' +
+           'Since there is no type of which all the above values are members, the type-variable constraint has been violated.\n');
```

Comment:
Ah, no. Never mind. For some reason I thought for a second Tuple needs both inner values to belong to the same type.

It still seems confusing to me though, that the output shows that we found a type for `["foo", true, 42]` (namely `Array ???`), but then we say that there's no types for that value.